File: java/src/main/java/br/com/rh/models/Employee.java
package br.com.rh.models;

import br.com.rh.enums.Role;

import java.math.BigDecimal;
import java.time.LocalDate;

public class Employee {
    private PersonalData personalData;
    private LocalDate lastReadjustDate;

    public Employee(String name, String cpf, Role role, BigDecimal salary) {
        this.personalData = new PersonalData(name, cpf, role, salary);
    }

    public void readjustSalary(BigDecimal newSalary) {
        this.personalData.setSalary(newSalary);
        this.lastReadjustDate = LocalDate.now();
    }

    public LocalDate getLastReadjustDate() {
        return lastReadjustDate;
    }

    public void promote(Role newRole) {
        this.personalData.setRole(newRole);
    }

    public PersonalData getPersonalData() {
        return personalData;
    }
}

File: typescript/src/models/employee.ts
import { Role } from "../enums/role.enum";
import { PersonalData } from "./personal-data";

export class Employee {
  private _personalData: PersonalData;
  private _lastReadjustDate: Date;

  constructor(name: string, cpf: string, role: Role, salary: number) {
    this._personalData = new PersonalData(name, cpf, role, salary);
    this._lastReadjustDate = new Date();
  }

  public readjustSalary(salary: number) {
    this._personalData.salary = salary;
    this._lastReadjustDate = new Date();
  }

  public promote(newRole: Role) {
    this.personalData.role = newRole;
  }

  get personalData() {
    return this._personalData;
  }

  get lastReadjustDate() {
    return this._lastReadjustDate;
  }
}
